week3/src/components/Game.jsx

```diff
@@ -0,0 +1,49 @@
+import React, { useRef, useState } from "react";
+import GameItem from "./GameItem";
+import gameInfo from "../utils/constant";
+import { Header, Main } from "../style/style";
+import vs from "../assets/vs.png";
+import crown from "../assets/crown.png";
+
+function Game() {
+  const winners = useRef([]);
+  const round = useRef(gameInfo.length / 2);
+  const [fighters, setFighters] = useState(gameInfo.sort(() => Math.random() - 0.5));
+  const [leftRound, setLeftRound] = useState(round.current);
+  const [win, setWin] = useState(false);
+
+  const handleClick = (fighter) => {
+    winners.current.push(fighter);
+    setLeftRound((current) => current - 1);
+    setFighters(fighters.slice(2));
```

Comment:
```suggestion
    setFighters(prevState => prevState.slice(2));
```
state값 업데이트할 때 state값에 직접 접근하기 보다 이런 식으로 최종 state인 prevState를 통해 업데이트 하는 편이 좋다고 합니다! 왜냐하면  `setFighters(fighters.slice(2));` 이런 식으로 업데이트 해버리면 setFighter를 중복해서 호출하게 되는 경우가 생긴다구 하네여

Reply:
> 

지나가는 길에 질문 한번 올리자면,, 혹시 왜 setFighters가 중복해서 호출되는건가요..??

Reply:
중복호출이라기보단 최신값 참조의 의미가 더 큽니다 ! !

Reply:
@choichoijin  어레,, 다시 읽어보니 제가 표현을 잘못한 것 같습미다.  중복해서 setState를 호출하는 상황이 생기면 저 안에 들어있는 fighters가 항상 최신값일 거라고 보장할 수 없다..라는 표현이 더 맞을 것 같아요!   팟짱님 말씀처럼  `가장 최신의 prevState를 받기 위해 저런 식으로 사용하는 편이 좋다 ! `가 맞는 것 같아여
https://www.delftstack.com/howto/react/react-setstate-prevstate/
이 아티클에서 문제 생기는 상황에 대해 자세히 다루고 있어서 한 번 읽어보시면 좋을 것 같슴다 헤헤..

Reply:
아하 setLeftRound의 경우에는 최신값을 current로 받아왔는데 setFighters에서는 그 부분을 놓친 것 같네요! 꼼꼼한 리뷰 감사합니다!